pkgs/by-name/xm/xmake/package.nix

```
@@ -1,28 +1,39 @@
 {
   lib,
   stdenv,
-  fetchurl,
+  fetchgit,
   nix-update-script,
+  versionCheckHook,
 }:
-stdenv.mkDerivation rec {
+
+stdenv.mkDerivation (finalAttrs: {
   pname = "xmake";
-  version = "2.9.9";
-  src = fetchurl {
-    url = "https://github.com/xmake-io/xmake/releases/download/v${version}/xmake-v${version}.tar.gz";
-    hash = "sha256-6SUFuDvJd2KG6ucZ1YvOp/8ld6/hLLXMsnnIHn28cC0=";
+  version = "3.0.0";
+
+  src = fetchgit {
+    url = "https://github.com/xmake-io/xmake.git";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-JY94DaSZ12pRrcCPxoPaECRA8rMK6/fNNyr/F8Htu8A=";
+    fetchSubmodules = true;
   };
```

This works for me.  Built on x86_64-linux.

```suggestion
  src = fetchFromGitHub {
    owner = "xmake-io";
    repo = "xmake";
    tag = "v${finalAttrs.version}";
    fetchSubmodules = true;
    hash = "sha256-JY94DaSZ12pRrcCPxoPaECRA8rMK6/fNNyr/F8Htu8A=";
  };
```

---

IIRC this is broken on macOS. I may test it later.

---

weird
now it worked
maybe it was something transient? idk
when I first tried it, it just didn't download any of the submodules